### PR TITLE
Clients can schedule new appointments

### PR DESCRIPTION
**Pull request by craigspaeth**
This adds the ability for clients to schedule new appointments based on finding one of their first coaching/therapy appointments and picking the same appt. type + practitioner based on that. It then shows a button "Schedule new coaching|therapy session" that leads to a page similar to the rescheduling page.

Apologies for the diff noise from the linter, the biggest updates occurring here are:

1. Updated the `client { appointments }` GraphQL query to be able to query any appts. after a time instead of just assuming upcoming appointments only
2. Refactored the rescheduler components by extracting a `Picker` component that's shared between rescheduling and scheduling pages
3. Added `newAppointment` to the view model which finds an old therapy/coaching appointment to base new appointments off of

cap

### CODE REVIEW

**Comment by lionsdwarf** [comment]
Couple small notes about query parameter names but otherwise LGTM!

**Comment by craigspaeth** [comment]
Thanks @lionsdwarf,  I don't see your comments about the query params though, did you forget to click Comment somewhere?

**Comment by lionsdwarf** [comment]
I forgot to "Submit Review"! You should be able to see the comments now.

**Comment by lionsdwarf** [comment]
Sorry about that !

File: apps/api/models/appointment.js [hunk 1]
```diff
@@ -144,17 +144,19 @@ export const fromCliniko = appt => ({
     )
 })
 
-export const findUpcomingForPatient = async (patientId, limit) => {
+export const findForPatient = async ({
+  createdTimeAfter,
```
[the remark this answers is not in the document]
Comment (craigspaeth) [reply apps/api/models/appointment.js]: I was basing the naming conventions off popular GraphQL services like GraphCMS & Prisma as noted [in the readme](https://github.com/findOctave/cortex#tech-architecture). Agreed, this is not totally natural/intuitive. It'd be nice if the field name being queried against mirrored the argument name—so I'll change to `createdTimeFrom`.

File: apps/api/models/appointment.js [hunk 2]
```diff
@@ -144,17 +144,19 @@ export const fromCliniko = appt => ({
...
+  first = 100,
```
[the remark this answers is not in the document]
Comment (craigspaeth) [reply apps/api/models/appointment.js]: Agreed, this is also odd but seems to be what common gql APIs use. 🤔 I'll change this b/c I'm not strongly opinionated on it. It'd be great if we had some kind of framework/basis for deciding these naming conventions if not trying to copy naming from other GQL APIs. Maybe we can just start building a playbook in the README for it... I'll add to this PR dedicating a README section to GraphQL conventions.

**Comment by craigspaeth** [comment]
Gracias! I'll make those updates and merge. Only [two hard things](https://martinfowler.com/bliki/TwoHardThings.html) right? 😬

**Comment by lionsdwarf** [comment]
Totally on board to help create the gql readme playbook! Great idea. This will inspire me toward more research before further ad-hoc "this doesn't feel right" commentary ;)

**Pull request merged**